src/Lean/Environment.lean

```diff
@@ -447,7 +458,14 @@ instance ModuleData.inhabited : Inhabited ModuleData :=
 @[extern 3 "lean_save_module_data"]
 constant saveModuleData (fname : @& String) (m : ModuleData) : IO Unit := arbitrary _
 @[extern 2 "lean_read_module_data"]
-constant readModuleData (fname : @& String) : IO ModuleData := arbitrary _
+constant readModuleData (fname : @& String) : IO (ModuleData × CompactedRegion) := arbitrary _
+
+/--
+  Free compacted regions of imports. No live references to imported objects may exist at the time of invocation; in
+  particular, `env` should be the last reference to any `Environment` derived from these imports. -/
+@[noinline, export lean_environment_free_regions]
+unsafe def Environment.freeRegions (env : Environment) : IO Unit :=
+env.header.regions.forM CompactedRegion.free
```

Comment:
It would be nice to have static guarantees (introspection tests?) here that this in fact frees all other fields of `env` first (in particular, `env` must  not be borrowed), but since the current compiler seems to be doing the right thing, that is sufficient for me for the time being.

Reply:
I can add basic support for introspection. A user-defined function that given the IR code checks whether there is a problem or not. There are different ways to enable user-defined checkers. I will add some very basic support for now.

Reply:
In the meantime, let's copy your comment to the file.

Reply:
done